v2/internal/genericarmclient/generic_client.go

```diff
@@ -93,7 +96,7 @@ func (client *GenericClient) ClientOptions() *arm.ClientOptions {
 	return client.opts
 }
 
-func (client *GenericClient) BeginCreateOrUpdateByID(ctx context.Context, resourceID string, apiVersion string, resource interface{}) (*PollerResponse, error) {
+func (client *GenericClient) BeginCreateOrUpdateByID(ctx context.Context, resourceID, apiVersion string, resource interface{}) (*PollerResponse, error) {
```

Comment:
Better to be explicit with types in function signatures - and given there are now several, consider putting them across several lines:

```suggestion
func (client *GenericClient) BeginCreateOrUpdateByID(
    ctx context.Context, 
    resourceID string, 
    apiVersion string, 
    resource interface{}) (*PollerResponse, error) {
```